### ISSUE TITLE
Fyne build and package do not work on mac

### ISSUE THREAD

**Issue by sk-** [fyne-io/fyne#4004]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Running `fyne build` for my project fails with the following error:

```
# fyne.io/fyne/v2/app
app_darwin.m:18:25: warning: 'UNUserNotificationCenter' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNUserNotificationCenter.h:39:12: note: 'UNUserNotificationCenter' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:18:6: note: annotate 'doSendNotification' with an availability attribute to silence this warning
app_darwin.m:19:5: warning: 'UNMutableNotificationContent' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNNotificationContent.h:97:12: note: 'UNMutableNotificationContent' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:19:5: note: enclose 'UNMutableNotificationContent' in an @available check to silence this warning
app_darwin.m:19:46: warning: 'UNMutableNotificationContent' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNNotificationContent.h:97:12: note: 'UNMutableNotificationContent' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:19:46: note: enclose 'UNMutableNotificationContent' in an @available check to silence this warning
app_darwin.m:26:5: warning: 'UNNotificationRequest' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNNotificationRequest.h:16:12: note: 'UNNotificationRequest' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:26:5: note: enclose 'UNNotificationRequest' in an @available check to silence this warning
app_darwin.m:26:39: warning: 'UNNotificationRequest' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNNotificationRequest.h:16:12: note: 'UNNotificationRequest' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:26:39: note: enclose 'UNNotificationRequest' in an @available check to silence this warning
app_darwin.m:37:5: warning: 'UNUserNotificationCenter' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNUserNotificationCenter.h:39:12: note: 'UNUserNotificationCenter' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:37:5: note: enclose 'UNUserNotificationCenter' in an @available check to silence this warning
app_darwin.m:37:41: warning: 'UNUserNotificationCenter' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNUserNotificationCenter.h:39:12: note: 'UNUserNotificationCenter' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:37:41: note: enclose 'UNUserNotificationCenter' in an @available check to silence this warning
app_darwin.m:41:5: warning: 'UNAuthorizationOptions' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNUserNotificationCenter.h:22:32: note: 'UNAuthorizationOptions' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:41:5: note: enclose 'UNAuthorizationOptions' in an @available check to silence this warning
app_darwin.m:41:38: warning: 'UNAuthorizationOptionAlert' is only available on macOS 10.14 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNUserNotificationCenter.h:22:32: note: 'UNAuthorizationOptions' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.11.0
app_darwin.m:41:38: note: enclose 'UNAuthorizationOptionAlert' in an @available check to silence this warning
# github.com/Topsort/aws-app
runtime.main_main·f: function main is undeclared in the main package

error building application: exit status 1
```

The reason for that error is that the build command is explicitly setting 10.11 to be the minimum version.

https://github.com/fyne-io/fyne/blob/1834d704a305f4729cce65213f5b5c3098e1d993/cmd/fyne/internal/commands/build.go#L223-L226

### How to reproduce

1. Create a new project on mac
2. Create a main.go file with the hello world (shown below)
3. Add any png with the name `Icon.png` to the development folder
3. run `fyne build` or `fyne package`

### Example code

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello World")

	w.SetContent(widget.NewLabel("Hello World!"))
	w.ShowAndRun()
}
```

### Fyne version

2.3.5

### Go compiler version

1.20.5

### Operating system and version

macOS 12.3 Monterey

**Comment by andydotxyz** [comment]
Can you please provide the full build command including any environment? We have 3 different notification code paths depending on the macOS version.

And yes, it does work normally - half our devs are on macOS and the CI tests it daily.

**Comment by sk-** [comment]
I found the issue, that error happens when there's no `main.go` file in the main directory (I had the app in `cmd/main.go`).

**Comment by andydotxyz** [comment]
Just FYI, it should not matter what you call the files - but you do need to run the build command from the application source directory (or specify the location in the build parameter).

**Comment by kichik** [comment]
For me the fix was downgrading from Go 1.25 to 1.24.
